website/static/js/pages/project-settings-page.js

```diff
@@ -128,8 +128,12 @@ $(document).ready(function() {
             var $elm = $(elm);
             formData[$elm.attr('name')] = $elm.is(':checked');
         });
+
+        var unchecked = checkedOnLoad.filter('#selectAddonsForm input:not(:checked)');
         var msgElm = $(this).find('.addon-settings-message');
-        $.ajax({
+        
+        var submit = function(){
+            $.ajax({
```

Comment:
As Nan notes, `$osf.postJSON` will let you greatly simplify this ajax call to:
`$osf.postJSON(url, payload)`

The callback could be passed in as a third argument: `{success: ...}`, or attached as a `done(...)` handler onto the promise returned by `postJSON`.